content/blog/grpc-web-interceptor.md

```diff
@@ -52,88 +36,58 @@ const UnaryInterceptor = function() {};
  *     invoker
  * @return {!Promise<!UnaryResponse<RESPONSE>>}
  */
-
 UnaryInterceptor.prototype.intercept = function(request, invoker) {};
 ```
 
-Unary/StreamInterceptor is designed as an interface with one method to be
-implemented. A dedicated class for interceptors will have better scalability
-since we usually want to implement it with some complex functionality. This will
-also make things a lot easier for gRPC-Web to be embedded in other client
-platforms.
+The `intercept()` method takes two parameters:
 
-The intercept method takes in two parameters:
+- A `request` of type [grpc.web.Request][]
+- An `invoker`, which performs the actual RPC when invoked
 
-- **[grpc.web.Request](https://github.com/grpc/grpc-web/blob/master/javascript/net/grpc/web/request.js)**:
-  A wrapper class of RequestMessage, MethodDescriptor, Metadata, CallOptions.
-- **Invoker function**: This function will make the actual RPC call and return
-  the response.
+The `StreamInterceptor` interface declaration is similar, except that the
+`invoker` return type is `ClientReadablaStream` instead of `Promise`. For
+implementation details, see [interceptor.js][].
 
-Besides grpc.web.Request, we’ve introduced another new class here:
+## What can I do with an interceptor?
 
-- **[grpc.web.UnaryResponse](https://github.com/grpc/grpc-web/blob/master/javascript/net/grpc/web/unaryresponse.js)**:
-  A wrapper class of response message, status and response metadata, so that we
-  could handle the response metadata(headers) via interceptors.
+Basically, an interceptor allows you to do the following:
 
-## Configure interceptors
+- Update the original gRPC request before it is passed along. For example, you
+  might inject extra information such as auth headers.
+- Manipulate the behavior of the original invoker function, such as bypassing
+  the call or updating the response before it is returned to the client.
```

Comment:
I suppose that it is possible to "bypass the RPC call" but how common and/or useful is that? Maybe we can stick to mentioning response updates?

Reply:
I think it could be common since people may want to use cached result instead? Another very common use case of intercepting response is implementing retry.

Reply:
Thanks, I think that mentioning use of cached results would be good. I'll add that.

Reply:
(I added a phrase about caching.)